README.md

```diff
@@ -91,97 +106,181 @@ cloning over SSH may provide a better experience (which requires
+options such as building a Swift-compatible LLDB, building the Swift Package
+Manager, building for iOS, running tests after builds, and more. It also
+supports presets, which you can define for common combinations of build options.
 
-    export PATH=/Applications/CMake.app/Contents/bin:$PATH
-    
-On Linux, if you have not already installed Swift's [development
-dependencies](#linux-dependencies), you can download and install the CMake
-package separately using the following command:
+View the inline help to find out more, in particular the section "Typical uses":
 
-    sudo apt-get install cmake
-    
+    utils/build-script -h
 
-#### Ninja
-[Ninja](https://ninja-build.org) is the current recommended build system
-for building Swift and is the default configuration generated by CMake. [Pre-built
-packages](https://github.com/ninja-build/ninja/wiki/Pre-built-Ninja-packages)
-are available for macOS and Linux distributions. You can also clone Ninja
-next to the other projects and it will be bootstrapped automatically:
+A basic command to build Swift with optimizations and run basic tests with
+Ninja:
 
-**Via HTTPS**
+    utils/build-script --release --test
 
-    git clone https://github.com/ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+### Development Environment
 
-**Via SSH**
+There are two primary build options to use with CMake, xcodebuild or Ninja.
+xcodebuild integrates nicely with Xcode.app, but Ninja is a bit faster and
+supports more environments.
 
-    git clone git@github.com:ninja-build/ninja.git && cd ninja
-    git checkout release
-    cat README
+To build using Xcode, run:
 
-#### Install via third-party packaging tool (macOS only)
+    utils/build-script --xcode
 
-**[Homebrew](https://brew.sh/)**
+The Xcode IDE can be used to edit the Swift source code, but it is not currently
+fully supported as a build environment for SDKs other than macOS. The generated
+Xcode project does not integrate with the test runner, but the tests can be run
+with the 'check-swift' target.
 
-    brew install cmake ninja
+If you need to work with other SDKs, you'll need to create a second build using
+Ninja.
 
-**[MacPorts](https://macports.org)**
+To build using Ninja, run:
 
-    sudo port install cmake ninja
+    utils/build-script --release
 
-### Building Swift
+If you are interested in compiler development it helps to build LLVM + Clang
+with optimizations enabled, but enable debug support in Swift. This is a good
+starting point since the debugger will work in the swift code base, and the
+tests will run faster.
 
-The `build-script` is a high-level build automation script that supports basic
-options such as building a Swift-compatible LLDB, building the Swift Package
-Manager, building for iOS, running tests after builds, and more. It also
-supports presets, which you can define for common combinations of build options.
+    utils/build-script --release-debuginfo --debug-swift
```

Comment:
I would also mention here that --debug-swift does not touch the stdlib. So for a fully debug experience with a release LLVM, you need to do:

./utils/build-script --release-debuginfo --debug-swift --debug-swift-stdlib.

Additionally, as an advanced feature, there is an option that you can use to compile just the typechecker in release. This is useful in reducing compile time if one only wants to debug /post-typechecking/. The option is:

--force-optimized-typechecker.

That being said, I am not 100% sure if this option is appropriate for this guide.